ldap/servers/slapd/back-ldbm/ldbm_entryrdn.c

```diff
@@ -1544,8 +1545,13 @@ _entryrdn_get_elem(entryrdn_db_ctx_t *ctx,
     /* Position cursor at the matching key */
     *elem = NULL;
     dbop = DBI_OP_MOVE_NEAR_DATA;
+
 retry_get:
+    if (*elem) {
+        slapi_ch_free((void**)&elem_data);
+    }
     rc = dblayer_cursor_op(&ctx->cursor, dbop, key, data);
+    elem_data = data->data;  /* save pointer to data so we can free it on a retry */
```

Comment:
would not be slapi_ch_free((void**)elem);  sufficient ?

Reply:
No because elem ptr is changed after we grab the original ptr address (when a retry occurs for a long rdn).  So we need to free the previous value (elem_data).  I'm reworking the variable name to be "prev_elem_data" so it's more clear.

Reply:
TBH this was a really tricky set of leaks to fix.  Took me a long time to finalize the fix.